src/main/java/moe/banana/jsonapi2/HasOne.java

```diff
@@ -19,6 +19,19 @@ public T get() throws ResourceNotFoundException {
         return (T) resource._doc.find(linkage);
     }
 
+    public T get(T defaultValue) {
+        try {
+            return get();
+        }
+        catch (ResourceNotFoundException e){
+            return defaultValue;
+        }
+    }
+
+    public boolean resourceExists(){
+        return resource._doc.contains(linkage);
+    }
+
```

Comment:
Umm... I'm wondering if we can borrow a name from *nix such as `follow a sym-link`. It helps if you'd like to just remove `resourceExists` and `Document.contains` from the commit so that I can make more changes to the semantic in both `HasOne` and `HasMany`.